cli/command/image/trust.go

```diff
@@ -301,8 +299,11 @@ func trustedPull(ctx context.Context, cli *command.DockerCli, repoInfo *registry
 		}
 
 		// If reference is not trusted, tag by trusted reference
-		if !r.reference.HasDigest() {
-			tagged, err := reference.WithTag(repoInfo, r.reference.String())
+		// TODO(aaronl): Is this code really necessary? Will notary ever
```

Comment:
I think removing it is fair, we can assert it is a digest somewhere else to prevent against unexpected results from the notary server.